common/src/main/java/com/wynntils/handlers/labels/LabelHandler.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.handlers.labels;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Handler;
+import com.wynntils.handlers.labels.event.LabelEvent;
+import com.wynntils.mc.event.SetEntityDataEvent;
+import com.wynntils.utils.mc.McUtils;
+import java.util.Optional;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.Entity;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LabelHandler extends Handler {
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onEntitySetData(SetEntityDataEvent event) {
+        Entity entity = McUtils.mc().level.getEntity(event.getId());
+
+        if (entity == null) return;
+
+        for (SynchedEntityData.DataValue<?> packedItem : event.getPackedItems()) {
+            if (Entity.DATA_CUSTOM_NAME.getId() == packedItem.id()) {
+                Optional<Component> value = (Optional<Component>) packedItem.value();
+                if (value.isEmpty()) continue;
+
+                WynntilsMod.postEvent(new LabelEvent(value.get()));
```

Comment:
So this is triggered when an entity gets a name, after spawning, or if it is renamed?

Just to check, the way we're e.g. detecting totems, by looking for armor stands and checking their name tag, it's not going to work in this case? Or would it, but you think this method is superior? 

I just find this quite opaque. Maybe because I don't fully understand this SynchedEntityData or PackedItems.

Reply:
We want to know when this happens, not that they exist. So event is better